crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -5346,8 +5346,8 @@ impl<'db> TypeInferenceBuilder<'db> {
                 todo_type!("`NotRequired[]` type qualifier")
             }
             KnownInstanceType::ClassVar => {
-                self.infer_type_expression(arguments_slice);
-                todo_type!("`ClassVar[]` type qualifier")
+                let ty = self.infer_type_expression(arguments_slice);
+                ty
```

Comment:
This is obviously not the full `ClassVar` support, but it seems reasonable to already do the trivial type-inference part of it.

Reply:
Per @carljm's comment in my Notion doc, though, we might actually want to "strip away" the `ClassVar` part of this in `infer_annotation_expression`, so that it's actually not possible for us to get a `ClassVar` inside `infer_type_expression` and the "lower" methods from that (such as this one).

Having said that, I think you are correct that this change will reduce false positives in the short term, and probably doesn't do any harm right now